src/Entity/AdherentEmailSubscriptionHistory.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace AppBundle\Entity;
+
+use Algolia\AlgoliaSearchBundle\Mapping\Annotation as Algolia;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Table(name="adherent_email_subscription_history")
+ * @ORM\Entity(repositoryClass="AppBundle\Repository\AdherentEmailSubscriptionHistoryRepository")
+ *
+ * @Algolia\Index(autoIndex=false)
+ */
+class AdherentEmailSubscriptionHistory
+{
+    /**
+     * @var int
+     *
+     * @ORM\Column(type="bigint")
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @var Adherent|null
+     *
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Adherent")
+     * @ORM\JoinColumn(onDelete="SET NULL")
```

Comment:
do we really need to store one subscription history without adherent (with NULL value for adherent_id)?

Reply:
Yes we want to keep them at all time to have consistent stats.